datasources/consumer.go

```diff
@@ -111,15 +122,113 @@ func (p *CDCConsumer) unmarshalEvent(value []byte, event *CDCEvent) error {
 }
 
 func (p *CDCConsumer) handleCDCEvent(event CDCEvent) error {
-	if !event.ShouldApplyUpdates() {
+	if event.ShouldApplyUpdates() {
+		p.logger.Infow("processing data sources event for user", "userid", event.FullDocument.UserID)
+		p.logger.Debugw("event being processed", "event", event)
+		if err := p.applyPatientDataSourcesUpdate(event); err != nil {
+			return err
+		}
+	} else {
 		p.logger.Debugw("skipping handling of event", "offset", event.Offset)
 		return nil
 	}
 
-	p.logger.Infow("processing data sources event for user", "userid", event.FullDocument.UserID)
-	p.logger.Debugw("event being processed", "event", event)
+	if p.containsNewDeviceIssues(event) {
```

Comment:
[nit] Should this block be inside the `if` block on line 125? Technically it is correct, but it took me a second to figure out why 1/2 of the functionality is in the `if` block and 1/2 is here. Or, move it all down here and restore the original `if` condition to exit early.